src/lib.rs

```diff
@@ -738,6 +831,27 @@ pub struct SignatureAlgorithm {
 	write_null_params :bool,
 }
 
+impl SignatureAlgorithm {
+	fn iter() -> std::slice::Iter<'static, &'static SignatureAlgorithm> {
+		static ALGORITHMS: [&'static SignatureAlgorithm; 4] = [
```

Comment:
style nit: please put the space before the colon.

Also, `'static` isn't needed in constants any more since... [1.17.0](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1170-2017-04-27). Also, a slice would be nicer than an array because then you don't have to update the length when you expand the list.